source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/input/ValidatedEditText.java

```diff
@@ -2,40 +2,102 @@
 
 import android.annotation.SuppressLint;
 import android.content.Context;
+import android.content.res.Resources;
 import android.graphics.Color;
 import android.graphics.PorterDuff;
+import android.util.TypedValue;
 import android.widget.EditText;
 import android.widget.LinearLayout;
 
-import io.adaptivecards.renderer.inputhandler.TextInputHandler;
+import io.adaptivecards.R;
 
 @SuppressLint("AppCompatCustomView")
 public class ValidatedEditText extends EditText
 {
-    public ValidatedEditText(Context context) {
+    public ValidatedEditText(Context context)
+    {
         super(context);
         setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
-        getBackground().setColorFilter(null);
+        m_isInvalid = false;
+        verifyIfUsingCustomInputs(context);
     }
 
-    public ValidatedEditText(Context context, int errorColor) {
+    public ValidatedEditText(Context context, int errorColor)
+    {
         this(context);
         m_errorColor = errorColor;
     }
 
+    /**
+     * Peeks the usingCustomInputs attribute in the style to verify if custom inputs are being used
+     * @param context Context to retrieve the value from
+     */
+    public void verifyIfUsingCustomInputs(Context context)
+    {
+        Resources.Theme theme = context.getTheme();
+        TypedValue isUsingCustomInputs = new TypedValue();
+
+        if (theme.resolveAttribute(R.attr.usingCustomInputs, isUsingCustomInputs, true))
+        {
+            m_isUsingCustomInputs = (isUsingCustomInputs.data != 0);
+        }
+        else
+        {
+            m_isUsingCustomInputs = false;
+        }
+    }
+
+    /**
+     * Updates the rendered edit text to display the error visual cues
+     * @param isValid Value signalizing if input has valid value
+     */
     public void setValidationResult(boolean isValid)
     {
-        if (isValid)
+        if (m_isUsingCustomInputs)
         {
-            // Change border color to original color
-            getBackground().setColorFilter(null);
+            // If using custom inputs, set the selector state for state_input_invalid as true
+            setInputInvalid(!isValid);
         }
         else
         {
-            // Change border to red
-            getBackground().setColorFilter(m_errorColor, PorterDuff.Mode.SRC_ATOP);
+            // If not using custom inputs, tint the whole input with attention color
```

Comment:
Instead of this conditional, would it be possible to always set the custom state, and include a default drawable in our sdk for EditText (just like the sample you created for the sample app here)?

That drawable would then be replaced by the host, if desired. This approach would allow us to remove the extra boolean flag.

Reply:
I would prefer to avoid having custom views for the controls as they may not match the remainder of the user application, if possible I would advice against modifying an application wide type of view just for our product